examples/produce_v3/python/streaming_produce_v3_main.py

```diff
@@ -0,0 +1,149 @@
+        self.__produce_records_thread = Thread(
+            target=self.__produce_records,
+            name="Producing records {}".format(topic)
+        )
+        self.__produce_records_thread.start()
+        # This will read the record-receipts from the http-connection, in full-duplex fashion.
+        self.__handle_record_receipts_thread = Thread(
+            target=self.__handle_record_receipts,
+            name="Handle record-receipts {}".format(topic)
+        )
+        self.__handle_record_receipts_thread.start()
+
+    # Produce a single record to the topic.
+    def produce(self, record: ProduceRecord):
+        self.__record_counter +=1 
+        self.__record_queue.put(record)
+
+    # Shutdown the producer.
+    def shutdown(self):
+        self.__close = True
+        self.__produce_records_thread.join()
+        self.__handle_record_receipts_thread.join()
+
+    def __record_generator(self):
+        while True:
+            try:
+                record =  self.__record_queue.get(timeout=5)
+            except queue.Empty:
+                print("No more records to produce, exiting __record_generator")
+                break
+            finally:
+                if(self.__close):
+                    break
+            print("Writing a record #%d with json %s" % (self.__record_counter, record.to_json()))
+            yield record.to_json()
+
+
+    def __produce_records(self):
+        self.connection = urllib3.connection.HTTPConnection(
+            host=self.__host,
+            port=self.__port
+        )
+        headers = urllib3.make_headers(
+        )
+        headers.update({
+            "Content-Type": "application/json",
+        })
+        print("Establishing connection with headers:", headers)
+        self.connection.request(
+            method='POST',
+            url="/v3/clusters/{}/topics/{}/records".format(
+                self.__cluster_id,
+                self.__topic,
+            ),
+            body=self.__record_generator(),
+            headers=headers,
+            chunked=True,
+        )
+        print("Done producing-records, exiting __produce_records")
+
+    def __handle_record_receipts(self):
+        print("Waiting for http-connection to be established.")
+        while not (hasattr(self, 'connection') and self.connection.sock is not None):
+            time.sleep(.1)
+        print("Connection established, will read responses.")
+
+        http_response_stream = HttpResponse(socket.SocketIO(self.connection.sock, "rb"))
+        print("Http-stream has status-code %d" % http_response_stream.status_code())
+        if http_response_stream.status_code() != 200:
+            raise Exception("Failed to produce records as recieved error with http status code %d, error %s" % (http_response_stream.status_code(), http_response_stream.body_string))
+
+        record_receipt_counter = 0
+        for chunk in http_response_stream:
+            if chunk == b'\r\n':
```

Comment:
@msn-tldr as stated in the design doc, https://confluentinc.atlassian.net/wiki/spaces/KREST/pages/1551573603/Produce+V3+API#ProduceV3API-Option1%3AHTTP%2F1.1ChunkedEncoding,

it is a workaround heartbeat so that it keeps TCP connection alive